Lib/test/test_codecs.py

```diff
@@ -3415,5 +3415,28 @@ def test_rot13_func(self):
             'To be, or not to be, that is the question')
 
 
+class CodecNameNormalizationTest(unittest.TestCase):
+    """Test the normalizestring function via codecs module"""
+    def test_normalized_encoding(self):
+        FOUND = (1, 2, 3, 4)
+        NOT_FOUND = (None, None, None, None)
+        def search_function(encoding):
+            if encoding == "aaa_8":
+                return FOUND
+            else:
+                return NOT_FOUND
+
+        codecs.register(search_function)
+        self.assertEqual(FOUND, codecs.lookup('AAA-8'))
```

Comment:
We should also test that a name in already normalized form is not changed:

```suggestion
        self.assertEqual(FOUND, codecs.lookup('aaa_8'))
        self.assertEqual(FOUND, codecs.lookup('AAA-8'))
```

Reply:
Thanks, make sense. Updated.